### PR TITLE
Coadd2d bugfix

### PR DESCRIPTION
**Pull request by jhennawi**
This small PR fixes a bug in the coadd2d when objects were not being identified. It also attempts to better streamline the extraction in coadd2d with existing methods in extract.py. It also adds some code to parse Gemini/GNIRS dither sequences.

### CODE REVIEW

File: pypeit/coadd2d.py [hunk 1]
```diff
@@ -627,14 +627,19 @@ def reduce(self, pseudo_dict, show=None, show_peaks=None, basename=None):
...
-                                                                                   model_noise=False, show_profile=show,
-                                                                                   show=show)
-
-        if self.find_negative and not parcopy['reduce']['extraction']['return_negative']:
-            sobjs.purge_neg()
+
+        skymodel_pseudo, objmodel_pseudo, ivarmodel_pseudo, outmask_pseudo, sobjs, _, _ = exTract.run(
+            global_sky_pseudo, prepare_extraction=False, model_noise=False, spat_pix=pseudo_dict['spat_img'])
+
+        # OLD CODE
```
Comment (profxj) [inline pypeit/coadd2d.py]: remove?

File: pypeit/extraction.py [hunk 2]
```diff
@@ -238,7 +238,7 @@ def initialise_slits(self, initial=False):
 #        # For echelle
 #        self.spatial_coo = self.slits.spatial_coordinates(initial=initial, flexure=self.spat_flexure_shift)
 
-    def extract(self, global_sky):
+    def extract(self, global_sky, model_noise=None, spat_pix=None):
```
Comment (profxj) [inline pypeit/extraction.py:241]: Add to doc string

File: pypeit/extraction.py [hunk 3]
```diff
@@ -307,12 +309,10 @@ def prepare_extraction(self, global_sky):
         msgs.info("Generating wavelength image")
         self.waveimg = self.wv_calib.build_waveimg(self.tilts, self.slits, spat_flexure=self.spat_flexure_shift)
 
-        # Set the initial and global sky
-        self.global_sky = global_sky
 
 
 
-    def run(self, global_sky, ra=None, dec=None, obstime=None):
+    def run(self, global_sky, prepare_extraction=True, model_noise=None, spat_pix=None, ra=None, dec=None, obstime=None):
```
Comment (profxj) [inline pypeit/extraction.py:342]: add to doc string

File: pypeit/scripts/coadd_2dspec.py [hunk 4]
```diff
@@ -140,6 +140,9 @@ def main(args):
         else:
             basename = args.basename
 
+
+        # TODO Heliocentric for coadd2d needs to be thought through. Currently turning it off.
```
Comment (profxj) [inline pypeit/scripts/coadd_2dspec.py:144]: Should we add a comment in `coadd2d.rst` ?
  seems like it

File: pypeit/spectrographs/gemini_gnirs.py [hunk 5]
```diff
@@ -244,6 +271,22 @@ def configuration_keys(self):
...
+    def pypeit_file_keys(self):
+        """
+        Define the list of keys to be output into a standard ``PypeIt`` file.
+
+        Returns:
+            :obj:`list`: The list of keywords in the relevant
+            :class:`~pypeit.metadata.PypeItMetaData` instance to print to the
+            :ref:`pypeit_file`.
+        """
+        #        pypeit_keys = super().pypeit_file_keys()
```
Comment (profxj) [inline pypeit/spectrographs/gemini_gnirs.py]: remove the commented lines

**Review by profxj — CHANGES_REQUESTED** [review]
Just a few doc requests
  thanks

File: pypeit/coadd2d.py [hunk 6]
```diff
@@ -593,7 +593,7 @@ def reduce(self, pseudo_dict, show=None, show_peaks=None, basename=None):
             objFind.show('image', image=pseudo_dict['imgminsky']*gpm.astype(float),
                        chname='imgminsky', slits=True, clear=True)
 
-        sobjs_obj, nobj = objFind.find_objects(sciImage.image, sciImage.ivar, show_peaks=show_peaks,
+        sobjs_obj, nobj = objFind.find_objects(sciImage.image, sciImage.ivar, show_peaks=show or show_peaks,
```
Comment (kbwestfall) [inline pypeit/coadd2d.py:601]: Please fill out the docstring for this function.  Include a description of both `show` and `show_peaks` and the different output when flag both or either.

File: pypeit/extraction.py [hunk 7]
```diff
@@ -278,16 +278,18 @@ def extract(self, global_sky):
             self.outmask = self.sciImg.fullmask
             self.skymodel = global_sky.copy()
         else:  # Local sky subtraction and optimal extraction.
+            model_noise_1 = (not self.bkg_redux) if model_noise is None else model_noise
```
Comment (kbwestfall) [inline pypeit/extraction.py]: Don't need the parentheses.

File: pypeit/scripts/coadd_2dspec.py (hunk 4 above)
Comment (kbwestfall) [reply pypeit/scripts/coadd_2dspec.py:144]: I agree.  Do we know if people are/will be coadding across multiple runs?

**Review by kbwestfall — APPROVED** [review]
Looks good.  Minor things from me, including some doc requests.

File: pypeit/coadd2d.py (hunk 6 above)
Comment (jhennawi) [reply pypeit/coadd2d.py:601]: Done.

File: pypeit/coadd2d.py (hunk 1 above)
Comment (jhennawi) [reply pypeit/coadd2d.py]: Done.

File: pypeit/extraction.py (hunk 2 above)
Comment (jhennawi) [reply pypeit/extraction.py:241]: Done.

File: pypeit/extraction.py (hunk 7 above)
Comment (jhennawi) [reply pypeit/extraction.py]: Thanks!

File: pypeit/extraction.py (hunk 3 above)
Comment (jhennawi) [reply pypeit/extraction.py:342]: Done.

File: pypeit/scripts/coadd_2dspec.py (hunk 4 above)
Comment (jhennawi) [reply pypeit/scripts/coadd_2dspec.py:144]: Well the coadd uses the individual waveimgs, so as long as heliocentric corrections are being done on those, I think this should work. However, I admittedly have not thought this through carefully which was the reason for the comment. Maybe we should make this an issue on GitHub.

File: pypeit/spectrographs/gemini_gnirs.py (hunk 5 above)
Comment (jhennawi) [reply pypeit/spectrographs/gemini_gnirs.py]: Done.

**Review by profxj — APPROVED** [review]
Generate the Issue or add a doc on the heliocentric,
 pass tests, and merge.  Thanks!

**Comment by badpandabear** [comment]
```
Unit Tests:
=========== 328 passed, 3844 warnings in 1057.71s (0:17:37) ==========
Dev suite:
--- PYPEIT DEVELOPMENT SUITE PASSED 148/148 TESTS (Masters ignored) ---
Testing Started at 2022-05-26T18:09:36.742114
Testing Completed at 2022-05-27T01:24:38.344687
Total Time: 7:15:01.602573

```

**Pull request merged**